src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -24,13 +23,17 @@ internal sealed class VisualStudioAnalyzer : IDisposable
         private readonly IAnalyzerAssemblyLoader _loader;
         private readonly string _language;
 
+        private readonly object _gate;
+
         private AnalyzerReference _analyzerReference;
-        private List<DiagnosticData> _analyzerLoadErrors;
+        private ImmutableArray<DiagnosticData> _analyzerLoadErrors;
 
         public event EventHandler UpdatedOnDisk;
 
         public VisualStudioAnalyzer(string fullPath, IVsFileChangeEx fileChangeService, HostDiagnosticUpdateSource hostDiagnosticUpdateSource, ProjectId projectId, Workspace workspace, IAnalyzerAssemblyLoader loader, string language)
         {
+            _gate = new object();
```

Comment:
why not?

Reply:
:bulb: I find it easier to read the single-instance gate initialization in this form:

https://github.com/dotnet/roslyn/blob/0a6c5618cf44d2dc0ac906959ff798e8eb2fa7e6/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs#L21

According to a quick git grep, both forms are used but initialization at the declaration site is used in 70-80% of cases.

Reply:
➡️ dotnet/roslyn-analyzers#1751

Reply:
I'll always take the one liner because then it's easier to confirm it's always being assigned, especially if somebody adds a second constructor.

Reply:
if we believe it must be a pattern. please put in an analyzer so that it is consistent on our code base. otherwise, sounds like personal preferences.

I do sometimes like fields being initialized in constructor since that's the only thing I need to look rather than looking around to find out how it got initialized. also, in debugger, when there are a lot of field initializer, much harder to follow since it goes to one stop to next as I steps.

Reply:
> Why the separate function? This seems to just make it harder to understand what Reset is doing because now I get to jump around.

so, in some cases, you feel the same then. since inlining requires one to jump around to see all initialization.